std/mmfile.d

```diff
@@ -331,8 +333,10 @@ class MmFile
                     char c = 0;
                     core.sys.posix.unistd.write(fd, &c, 1);
                 }
-                else if (prot & PROT_READ && size == 0)
+                else if (prot & PROT_READ && size == 0) {
```

Comment:
It seems like there is a fair amount of code duplication between the two codepath that you modified. Is this hard to refactor? If not, this refactoring should come as a first step.